server/src/main/java/io/littlehorse/common/model/metadatacommand/subcommand/PutPrincipalRequestModel.java

```diff
@@ -98,8 +98,9 @@ public Principal process(MetadataCommandExecution context) {
 
             toSave.setCreatedAt(oldPrincipal.getCreatedAt());
         }
-
-        if (perTenantAcls.isEmpty()) {
+        boolean canWriteAdminPrincipals = requester.isAdmin();
+        if (perTenantAcls.isEmpty()
+                && !(canWriteAdminPrincipals && !globalAcls.getAcls().isEmpty())) {
```

Comment:
I am confused—what is the purpose of this line? And why do we want to throw the error?

Reply:
Basically `!(canWriteAdminPrincipals && !globalAcls.getAcls().isEmpty())` this condition verifies that if there is no per tenant acls at least it contains global acls, and it verifies that the requester is an admin principal

Reply:
Ok, we will allow store principal without any privilege.